torch/_functorch/partitioners.py

```diff
@@ -1403,9 +1406,13 @@ def _optimize_runtime_with_given_memory(
         return ilp_knapsack(memory, runtimes, max_memory)
     elif SOLVER == "dp":
         return dp_knapsack(memory, runtimes, max_memory)
-    elif callable(SOLVER):
-        saved_node_idx, recomp_node_idx = SOLVER(
-            memory, joint_graph, max_memory, node_info, all_recomputable_banned_nodes
```

Comment:
Why are you removing this functionality of setting `SOLVER=some_callable` here?

Can we use this code path for commercial solvers instead?  That seems better than having a non-working solver checked in.  Rather than patching `backwards_pass_aware_graph_solver` you could patch `SOLVER`.

Reply:
Discussed offline. Planning to change the implementation here to include all logic in OSS.